ts/components/wallet/card/index.tsx

```diff
@@ -77,18 +78,18 @@ type ReduxMappedStateProps = Readonly<{
 }>;
 
 type ReduxMappedDispatchProps = Readonly<{
-  setFavoriteCard: (item: Option<CreditCardId>) => void;
+  setFavoriteCard: (item: Option<number>) => void;
 }>;
 
 export type CardProps = Readonly<{
-  item: CreditCard;
+  item: Wallet;
   navigation: NavigationScreenProp<NavigationState>;
   menu?: boolean;
   favorite?: boolean;
   lastUsage?: boolean;
   whiteLine?: boolean;
   logoPosition?: LogoPosition;
-  mainActionNavigation?: string;
+  mainAction?: (cardId: number) => void;
```

Comment:
why is `mainAction` optional? what is the case when we ignore the clicks on the cards?

Reply:
e.g. [here](https://projects.invisionapp.com/d/main/default/#/console/13806972/287731306/preview) and [here](https://projects.invisionapp.com/d/main/default/#/console/13806972/287731271/preview)